### ISSUE TITLE
ClearTasks retains old task info data

### ISSUE THREAD

**Issue by dchaboya** [jsGanttImproved/jsgantt-improved#226]
When i want to "refresh" my gantt chart with new data coming from a database, i do the following: 1) call ClearTasks, 2) call Draw, 3) fetch data and add it to the gantt chart using AddTaskItemToObject, 4) and finally call Draw.  When i do this, data in the task info popup is not refreshed.  I have to refresh the page itself to see the changes in that data.

**Comment by nidu** [comment]
Just want to say that i recreate entire component on every change to be sure i'm in sync. Works fast enough.

**Comment by mariohmol** [comment]
@nidu i'm making the same here.. but do you have any ideas why this happens?

**Comment by nidu** [comment]
@dchaboya i just tried to add following function to the demo page and called it. Get the same issue with the tooltip.

```
function doTheThing(postfix) {
  const jsonLocal = jsonObj.map(node => ({...node, pName: node.pName + postfix}));
  g.ClearTasks();
  g.Draw();
  setTimeout(() => {
    for (const item of jsonLocal) {
      g.AddTaskItemObject(item); 
    }
    g.Draw();
  }, 1000);
```

What if you hover over another task and then hover over initial one again? It updates tooltip content for me. That's probably because tooltip is not updated if it hovers again over item with the same ID, which is the case here. We can fix it by resetting `vToolCont` `showing` attribute like [here](https://github.com/nidu/jsgantt-improved/commit/320e48ee666c08caa834537e88cdd53965a6c1a8)

**Comment by mariohmol** [comment]
this will force it to remake the html every time he hovers, right? i think is a good idea!

**Comment by nidu** [comment]
Yeap, when hovering away and back to the same task. I'm not sure if resetting showing can affect anything badly though (doubt it).

**Comment by mariohmol** [comment]
i guess just performance issue.. but will not change that much.. u want to fill a PR or I make it here?

**Comment by nidu** [comment]
I don't think performance would be an issue here since we update a tiny part of HTML! Moreover source HTML element is already created. Probably we can increase performance a bit turning `pContents.innerHTML` to string once for this place in `events.ts`:

```
pGanttChartObj.vTool.vToolCont.innerHTML = pContents.innerHTML;
```

Also no need to store `pContents` HTML element.

I've made a pull request, since i already did the change

**Comment by mariohmol** [comment]
released on v2.3.7

thanks @nidu
